library-best-practices/JLBP-17.md

```diff
@@ -0,0 +1,30 @@
+[JLBP-17] Coordinate Major Version Adoption
+-------------------------------------------
+
+It is clearly undesirable for any particular library dependency tree to contain
+conflicts. Unfortunately, for any consumer of such a dependency tree that tries
+to use the highest versions available of all libraries contained in it, there
+are conflicts as soon as a breaking change is introduced. Thus, breaking changes
+need to be propagated as quickly as possible so that such consumers can continue
+using the highest versions.
+
+Perform the rollout in this manner:
+
+1. Decide whether to break the feature as an atomic change or to perform the
+   change in two passes (deprecate then delete).
+   - If the feature is used by stable code in other libraries, you must use two
+     passes (as per [JLBP-7](JLBP-7.md)).
+   - If there does not appear to be usage of the code in other libraries, an
+     in-place breakage may be ok.
+   - If the feature is used by unstable code or stable code having little usage
+     itself, prefer to use two passes if possible, with in-place breakage only
+     if two passes is judged to be too costly.
+2. Make sure that consuming libraries are prepared for the breakage.
+   - In the case of in-place breakage, have a PR prepared and approved in the
+     consuming libraries that switches from the old surface to the new surface.
```

Comment:
This doesn't work. The consuming won't approve such a PR until the continuous build passes, which it won't at this stage.

Reply:
I mean provisionally approved - formal approval would wait for the actual roll out. I will clarify.